src/main/java/org/kiwiproject/registry/consul/client/ConsulRegistryClient.java

```diff
@@ -92,9 +102,29 @@ private ServiceInstance fromCatalogService(CatalogService catalogService) {
                 .version(metadata.get("version"))
                 .status(ServiceInstance.Status.UP)
                 .ip(metadata.get("ipAddress"))
+                .metadata(serviceMetadata)
                 .build();
     }
 
+    private Map<String, String> filterMetadata(Map<String, String> metadata) {
+        return metadata.entrySet().stream()
+                .filter(entry -> !METADATA_EXCLUDED_KEYS.contains(entry.getKey()))
+                .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private void addTagsToMetadata(Map<String, String> metadata, List<String> tags) {
+        var filteredTags = tags.stream()
+                .filter(tag -> !TAGS_EXCLUDED.contains(tag))
+                .collect(toList());
+
+        if (isNotNullOrEmpty(filteredTags)) {
+            filteredTags.forEach(tag -> {
+                var tagSplit = tag.split(":");
+                metadata.put(tagSplit[0], tagSplit[1]);
```

Comment:
Aer Consul tags _always_ in the format `x:y`? If not we will run into `ArrayIndexOutOfBoundsException`

Reply:
Apparently according to @terezivy no they aren't.  We are always setting ours that way but that is not guaranteed.  I will add a check.